src/r8/r8.targets

```diff
@@ -0,0 +1,119 @@
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <UsingTask AssemblyFile="..\..\bin\Build$(Configuration)\xa-prep-tasks.dll" TaskName="Xamarin.Android.BuildTools.PrepTasks.DownloadUri" />
+  <UsingTask AssemblyFile="..\..\bin\Build$(Configuration)\xa-prep-tasks.dll" TaskName="Xamarin.Android.BuildTools.PrepTasks.SetEnvironmentVariable" />
+  <UsingTask AssemblyFile="..\..\bin\Build$(Configuration)\Xamarin.Android.Tools.BootstrapTasks.dll" TaskName="Xamarin.Android.Tools.BootstrapTasks.UnzipDirectoryChildren" />
+
+  <!-- public targets -->
+
+  <PropertyGroup>
+    <BuildDependsOn>
+      _DownloadDepotTools;
+      _UnzipDepotTools;
+      _SetDepotToolsEnvironment;
+      _BootstrapDepotTools;
+      _BuildR8;
+      _CopyR8;
+    </BuildDependsOn>
+    <CleanDependsOn>
+      _SetDepotToolsEnvironment;
+      _CleanR8;
+      _CleanDepotTools;
+    </CleanDependsOn>
+  </PropertyGroup>
+  <Target Name="Build" DependsOnTargets="$(BuildDependsOn)" />
+  <Target Name="Clean" DependsOnTargets="$(CleanDependsOn)" />
+
+  <!-- 
+    depot_tools, a set of git extensions from Chromium
+    http://commondatastorage.googleapis.com/chrome-infra-docs/flat/depot_tools/docs/html/depot_tools_tutorial.html
+  -->
+
+  <PropertyGroup>
+    <_DepotToolsZip>..\..\bin\Build$(Configuration)\depot_tools.zip</_DepotToolsZip>
+    <_PathToDepotTools>$([System.IO.Path]::GetFullPath('$(MSBuildThisFileDirectory)..\..\bin\Build$(Configuration)\depot_tools'))</_PathToDepotTools>
+  </PropertyGroup>
+
+  <Target Name="_DownloadDepotTools">
+    <DownloadUri
+        SourceUris="https://storage.googleapis.com/chrome-infra/depot_tools.zip"
```

Comment:
Should this URL be overridable?

Should we download this into `$HOME\android-archives` with some form of "version embedding", like the other android-archives logic, so that if we `git clean` we don't need to *re*-download this file? Though I do wonder how to perform the "version embedding", short of using some HTTP header.  `etag` looks interesting, as do some of the `x-google` headers:

```bash
$ curl -D dp.h -o depot_tools.zip 'https://storage.googleapis.com/chrome-infra/depot_tools.zip'
$ cat dp.h
HTTP/2 200 
x-guploader-uploadid: AEnB2Uo5obu-HRsZKe-A1jE-D4Cwhrf_tn9qqXiLT-VOULD3N8EKgaU3EkqPYYocij79Tsix8I2U1bnCbBmJnGmtETx6bn_LJQ
expires: Tue, 23 Oct 2018 02:37:02 GMT
date: Tue, 23 Oct 2018 01:37:02 GMT
cache-control: public, max-age=3600
last-modified: Mon, 22 Oct 2018 19:16:25 GMT
etag: "22387ac2ff531d150322d214ac4d0d90"
x-goog-generation: 1540235785605087
x-goog-metageneration: 1
x-goog-stored-content-encoding: identity
x-goog-stored-content-length: 21032694
content-type: application/zip
x-goog-hash: crc32c=/k0K7g==
x-goog-hash: md5=Ijh6wv9THRUDItIUrE0NkA==
x-goog-storage-class: STANDARD
accept-ranges: bytes
content-length: 21032694
server: UploadServer
alt-svc: quic=":443"; ma=2592000; v="44,43,39,35"
```

We could always update `<DownloadUri/>` to "know about" these headers and optionally embed them into the downloaded filename... which would promptly fail if we allow changing the target URL, but ¯\_(ツ)_/¯?

Reply:
> Should this URL be overridable?

Yeah I'll add that.

> Should we download this into `$HOME\android-archives` with some form of "version embedding"

The [instructions](https://commondatastorage.googleapis.com/chrome-infra-docs/flat/depot_tools/docs/html/depot_tools_tutorial.html#_setting_up) clearly show how to git clone & build `depot_tools` on Linux & Mac. But Windows it says "download this file", which I assumed means "don't build this on Windows! it won't work!". When I took this over, I was having trouble getting `depot_tools` to build on Windows, until I found this doc and used the download instead.

But I think I should still rework this to go in `~/android-archives` and `~/android-toolchain` like everything else. It should get moved to `android-toolchain` to be like cmake, ant, etc. and how they get downloaded and unzipped. I'll see about using `x-goog-hash` w/ `DownloadUri`, too. I think `etag` is a general HTTP cache expiration that is more precise than `expires`.